packages/expo-modules-core/common/cpp/EventEmitter.cpp

```diff
@@ -0,0 +1,140 @@
+#include "JSIUtils.h"
+#include "EventEmitter.h"
+
+namespace expo::EventEmitter {
+
+#pragma mark - Listeners
+
+void Listeners::add(jsi::Runtime &runtime, std::string eventName, const jsi::Function &listener) noexcept {
+  listenersMap[eventName].push_back(jsi::Value(runtime, listener));
```

Comment:
```suggestion
  listenersMap[eventName].emplace_back(runtime, listener);
```
I believe that adding to the list of complex objects in C++ is more efficient by reducing the number of copies that must be made under the hood. Although it may not be that significant, I think it is a more natural way of using collections in C++.